osu.Framework/Input/Handlers/Mouse/SDL2RelativeMouseHandler.cs

```diff
@@ -0,0 +1,139 @@
+
+using System.Diagnostics;
+using osu.Framework.Bindables;
+using osu.Framework.Input.StateChanges;
+using osu.Framework.Platform;
+using osu.Framework.Statistics;
+using osuTK;
+using osuTK.Input;
+
+namespace osu.Framework.Input.Handlers.Mouse
+{
+    public class SDL2RelativeMouseHandler : InputHandler, IHasCursorSensitivity, INeedsMousePositionFeedback
+    {
+        public BindableDouble Sensitivity { get; } = new BindableDouble(1) { MinValue = 0.1, MaxValue = 10 };
+
+        public override bool IsActive => true;
+
+        public override int Priority => 0;
+
+        private SDL2DesktopWindow window;
+
+        private Vector2? lastPosition;
+
+        private IBindable<bool> isActive;
+
+        /// <summary>
+        /// Whether a non-relative mouse event has ever been received.
+        /// This is used as a starting location for relative movement.
+        /// </summary>
+        private bool absolutePositionReceived;
+
+        public override bool Initialize(GameHost host)
+        {
+            if (!(host.Window is SDL2DesktopWindow desktopWindow))
+                return false;
+
+            window = desktopWindow;
+
+            isActive = window.IsActive.GetBoundCopy();
+            isActive.BindValueChanged(_ => updateRelativeMode());
+
+            Enabled.BindValueChanged(enabled =>
+            {
+                updateRelativeMode();
+
+                if (enabled.NewValue)
+                {
+                    window.MouseMove += handleMouseMove;
+                    window.MouseMoveRelative += handleMouseMoveRelative;
+                    window.MouseDown += handleMouseDown;
+                    window.MouseUp += handleMouseUp;
+                    window.MouseWheel += handleMouseWheel;
+                }
+                else
+                {
+                    window.MouseMove -= handleMouseMove;
+                    window.MouseMoveRelative -= handleMouseMoveRelative;
+                    window.MouseDown -= handleMouseDown;
+                    window.MouseUp -= handleMouseUp;
+                    window.MouseWheel -= handleMouseWheel;
+                }
+            }, true);
+
+            return true;
+        }
+
+        public void FeedbackMousePositionChange(Vector2 position)
+        {
+            if (!Enabled.Value)
+                return;
+
+            updateRelativeMode();
+
+            if (window.RelativeMouseMode)
+            {
+                // store the last mouse position to propagate back to the host window manager when exiting relative mode.
+                lastPosition = position;
+            }
+
+            // handle the case where relative / raw input is active, but the cursor may have exited the window
+            // bounds and is not intended to be confined.
+            if (window.RelativeMouseMode && !window.CursorConfined)
+            {
+                if (position.X < 0 || position.Y < 0 || position.X > window.Size.Width || position.Y > window.Size.Height)
```

Comment:
sounds like it should be:
```suggestion
                if (position.X < 0 || position.Y < 0 || position.X >= window.Size.Width || position.Y >= window.Size.Height)
```